### DISCUSSION TITLE
Movement speed modifier

### DISCUSSION THREAD

**Discussion by HuHuMac** [acemod/ACE3#9323] [Q&A]
Hello everyone (I am French, sorry for my English), 
I searched everywhere and could not find what I need so I came here in order to ask you what seems to be an easy thing to everyone on the internet EXCEPT ME. 

I would like to enhance the movement speed of **ONE specific unit** so I use "this setAnimSpeedCoef 2". 
The unit is now going 2 times faster, great. BUT when I use ace with advanced stamina (cause I would like all other units to be with normal speed and use ace stamina) then suddenly it does not work anymore, ace advanced stamina seems to take over what i did put in the init of the unit that I wanted to be faster than others. I repeat I don't want to remove ACE advanced stamina, I just want certain units to not use it so I can give them more movement speed or remove stamina for them specifically. 

I don't see any solutions for my problem, please help me

**Comment by PabstMirror** [comment]
`ace_advanced_fatigue_setAnimExclusions pushBack "myMod";`
will block adv fatigue from changing setAnimSpeedCoef 

but it also looks like ace medical treatment may also change it
there might be a better way but this should restore the anim on your unit after treatment completes

```
["ace_common_setAnimSpeedCoef", {
    params ["_object", "_set"];
    if (_object == specialUnit && {_set == 1}) then {
        [{_this setAnimSpeedCoef 2}, _object] call CBA_fnc_execNextFrame
    };
}] call CBA_fnc_addEventHandler;
```

**Reply by HuHuMac** [reply]
Hello, unfortunately this doesn't seem to work. 
I did put what you wrote inside the init of the unit, it does not work, am I missing something ? I tried adding the command I used earlier but it doesn't do the trick. 
I run twice as fast for 1 sec, then the stamina bar shows up which means ace loaded and is now working, it puts me back to normal speed
